src/modules/previewpane/FileActionsMenu/FileActionsMenu.Plugins/PowerToys.FileActionsMenu.Plugins.FileProperties/UnblockFiles.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using FileActionsMenu.Interfaces;
+using FileActionsMenu.Ui.Helpers;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+
+namespace PowerToys.FileActionsMenu.Plugins.FileProperties
+{
+    internal sealed class UnblockFiles : IAction
+    {
+        private string[]? _selectedItems;
+
+        public string[] SelectedItems { get => _selectedItems.GetOrArgumentNullException(); set => _selectedItems = value; }
+
+        public string Title => "Unblock files";
+
+        public IAction.ItemType Type => IAction.ItemType.SingleItem;
+
+        public IAction[]? SubMenuItems => null;
+
+        public int Category => 8;
+
+        public IconElement? Icon => new FontIcon { Glyph = "\ue785" };
+
+        public bool IsVisible => SelectedItems.Any(file => File.Exists(file + ":Zone.Identifier"));
```

Comment:
May a damn question, maybe a good one. What happens if one selected item is a folder? Need for catching that case?

Reply:
No. Nothing happens then.